packages/ui-components/src/BrowseFileDialog.tsx

```diff
@@ -31,32 +31,80 @@ export interface IBrowseFileDialogOptions {
   multiselect?: boolean;
   includeDir?: boolean;
   acceptFileOnDblClick?: boolean;
+  rootPath?: string;
+  startPath?: string;
+}
+
+interface IBrowseFileBreadCrumbsOptions extends BreadCrumbs.IOptions {
+  rootPath?: string;
+}
+
+/**
+ * Breadcrumbs widget for browse file dialog body.
+ */
+class BrowseFileDialogBreadcrumbs extends BreadCrumbs {
+  model: any;
+  rootPath: string;
+
+  constructor(options: IBrowseFileBreadCrumbsOptions) {
+    super(options);
+    this.model = options.model;
+    this.rootPath = options.rootPath;
+  }
+
+  protected onUpdateRequest(msg: any): void {
+    super.onUpdateRequest(msg);
+    const contents = this.model.manager.services.contents;
+    const localPath = contents.localPath(this.model.path);
+
+    // if 'rootPath' is defined prevent navigating to it's parent/grandparent directories
+    if (localPath && this.rootPath && localPath.indexOf(this.rootPath) === 0) {
+      const breadcrumbs = document.querySelectorAll(
+        '.elyra-browseFileDialog .jp-BreadCrumbs > span[title]'
+      );
+
+      breadcrumbs.forEach((crumb: Element): void => {
+        if ((crumb as HTMLSpanElement).title.indexOf(this.rootPath) === 0) {
+          crumb.className = crumb.className.replace(
+            'elyra-BreadCrumbs-',
+            'jp-BreadCrumbs-'
+          );
```

Comment:
is there not a better way of handling this than changing the class? Like adding a class and making the css target more specific? (editing html like this just feels hack-y to me, personal opinion)

Reply:
`BreadCrumbs` processes click events based off of the `className` for the DOM node.

the approach taken here is to prevent the click event by changing the `className` (so it won't get processed). 

initially, i did try to override the `handleEvent` in `BreadCrumbs` as a better way to prevent handling of specific click events. unfortunately, `BreadCrumbs` is not really extensible and typescript prevents accessing the necessary private functions (e.g., `_evtClick`) and private variables (e.g., `_crumbs`) declared in `BreadCrumbs`.

another option is to re-implement the entire breadcrumbs widget and duplicate/copy over most of the existing `BreadCrumbs` code, but i suspect this is overkill (just to be able to disable couple of links)

Reply:
`BreadCrumbs` processes click events based off of the `className` for the DOM node. adding a class and making CSS target more specific won't prevent handling of the click event

the approach taken here is to prevent the click event by changing the `className` (so it won't get processed). 

in any case, i have updated the PR just added a new class instead (then use css to prevent the click event)